config/eslint-config/variants/base-ts.config.js

```diff
@@ -0,0 +1,59 @@
+import BaseJSConfig from "./base-js.config.js";
+import ts from "typescript-eslint";
+import { resolve } from "path";
+
+const tsconfigPath = resolve(process.cwd(), "tsconfig.json");
+
+const RecommendedTSConfig = ts.configs.recommendedTypeChecked.map((config) => ({
+  ...config,
+  files: ["**/*.ts", "**/*.tsx"],
+  languageOptions: {
+    ...config.languageOptions,
+    parserOptions: {
+      project: tsconfigPath,
+    },
+  },
+}));
+
+/** @type {import("eslint").Linter.Config} */
+const BaseTSConfig = {
+  files: ["**/*.ts"],
+  plugins: {
+    "@typescript-eslint": ts.plugin,
+  },
+  languageOptions: {
+    parser: ts.parser,
+    parserOptions: {
+      sourceType: "module",
+      project: tsconfigPath,
+    },
+  },
+  rules: {
+    "@typescript-eslint/class-literal-property-style": "warn",
+    "@typescript-eslint/no-duplicate-enum-values": "warn",
+    "@typescript-eslint/prefer-for-of": "warn",
+    "@typescript-eslint/consistent-type-definitions": ["error", "interface"],
+    "@typescript-eslint/no-unused-vars": [
+      "warn",
+      {
+        ignoreRestSiblings: true,
+        argsIgnorePattern: "^_",
+        varsIgnorePattern: "^_",
+      },
+    ],
+    // Migrated from old typescript-eslint config
+    "@typescript-eslint/ban-ts-comment": "off",
+    "@typescript-eslint/ban-ts-ignore": "off",
+  },
+};
+
+export default ts.config(
+  ...BaseJSConfig,
+  ...RecommendedTSConfig,
```

Comment:
Shouldn't the recommended overwrite the base?

Reply:
1. the target different file extensions
2. if the file extensions are the same and different rules are targeted they will be merged